src/libsyntax/parse/mod.rs

```diff
@@ -424,47 +424,40 @@ mod tests {
                 string_to_stream("macro_rules! zip (($a)=>($a))".to_string()).trees().collect();
             let tts: &[TokenTree] = &tts[..];
 
-            match (tts.len(), tts.get(0), tts.get(1), tts.get(2), tts.get(3)) {
-                (
-                    4,
-                    Some(&TokenTree::Token(Token {
-                        kind: token::Ident(name_macro_rules, false), ..
-                    })),
-                    Some(&TokenTree::Token(Token { kind: token::Not, .. })),
-                    Some(&TokenTree::Token(Token { kind: token::Ident(name_zip, false), .. })),
-                    Some(&TokenTree::Delimited(_, macro_delim, ref macro_tts)),
-                )
+            match tts {
+               &[TokenTree::Token(Token {kind: token::Ident(name_macro_rules, false), ..}),
```

Comment:
```suggestion
               &[TokenTree::Token(Token { kind: token::Ident(name_macro_rules, false), ..}),
```

Reply:
Done. 
(I did not ran rustfmt to prevent noise in changes)